community/opendkim/opendkim.post-install

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+selector=$(grep ^Selector /etc/opendkim/opendkim.conf | awk '{print $2}')
+keyfile=$(grep ^KeyFile /etc/opendkim/opendkim.conf | awk '{print $2}')
+domain=$(grep ^Domain /etc/opendkim/opendkim.conf | awk '{print $2}')
+
+[ "$keyfile" ] || exit 0
+
+if [ ! -f $keyfile ]; then
+	keyname=${keyfile%.private}
+	install -dm750 -g mail ${keyfile%/*}
+	openssl genrsa -out $keyfile 1024 2>/dev/null
+	openssl rsa -in $keyfile -pubout -out $keyname.public -outform PEM 2>/dev/null
+	chmod 600 $keyfile
+	chown opendkim $keyfile
+	[ "$domain" ] || exit 0
+	echo -en "$selector._domainkey IN TXT (\n\"v=DKIM1; k=rsa;\"\n\"p=" > $keyname.txt
+	grep -v ^- $keyname.public | tr -d '\n' >> $keyname.txt
+	echo -e "\"\n) ; --- DKIM key $selector for $domain" >> $keyname.txt
+	rm $keyname.public
+fi
+
+exit 0
```

Comment:
I do it the same as in other distros. Private key file must exists to start service.
The standalone perl-based key generation script 'opendkim-genkey' is on opendkim-utils subpackage.
Maybe the better way to move it to post-upgrade step to regenerate keys with "apk fix opendkim"